2024-01-01 10:10:33.497174+05:30 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "MIGRATION_MODEL" not found (this database is empty); SQL statement:
SELECT ID, VERSION FROM MIGRATION_MODEL ORDER BY UPDATE_TIME DESC [42104-224]
2024-01-01 10:10:36.762529+05:30 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "DATABASECHANGELOG" not found (this database is empty); SQL statement:
SELECT COUNT(*) FROM PUBLIC.DATABASECHANGELOG [42104-224]
2024-01-01 10:10:37.042387+05:30 jdbc[4]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "DATABASECHANGELOGLOCK" not found (this database is empty); SQL statement:
SELECT COUNT(*) FROM PUBLIC.DATABASECHANGELOGLOCK [42104-224]
2024-01-01 10:10:37.067383+05:30 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "DATABASECHANGELOG" not found; SQL statement:
SELECT COUNT(*) FROM PUBLIC.DATABASECHANGELOG [42102-224]
2024-01-03 10:49:20.809294+05:30 database: flush
org.h2.message.DbException: General error: "org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@394e00e7 failed; length 4096 at 8192 [2.2.224/2]" [50000-224]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.FileStore.storeBuffer(FileStore.java:1527)
	at org.h2.mvstore.FileStore.lambda$serializeAndStore$3(FileStore.java:1432)
	at org.h2.mvstore.FileStore.submitOrRun(FileStore.java:1401)
	at org.h2.mvstore.FileStore.serializeAndStore(FileStore.java:1432)
	at org.h2.mvstore.FileStore.lambda$storeIt$2(FileStore.java:1376)
	at org.h2.mvstore.FileStore.submitOrRun(FileStore.java:1401)
	at org.h2.mvstore.FileStore.storeIt(FileStore.java:1375)
	at org.h2.mvstore.MVStore.storeNow(MVStore.java:843)
	at org.h2.mvstore.MVStore.store(MVStore.java:808)
	at org.h2.mvstore.MVStore.commit(MVStore.java:778)
	at org.h2.mvstore.MVStore.commit(MVStore.java:770)
	at org.h2.mvstore.tx.TransactionStore.close(TransactionStore.java:365)
	at org.h2.mvstore.db.Store.close(Store.java:349)
	at org.h2.engine.Database.closeOpenFilesAndUnlock(Database.java:1293)
	at org.h2.engine.Database.<init>(Database.java:401)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:343)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:125)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@394e00e7 failed; length 4096 at 8192 [2.2.224/2]" [50000-224]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 34 more
Caused by: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@394e00e7 failed; length 4096 at 8192 [2.2.224/2]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:478)
	at org.h2.mvstore.SingleFileStore.writeFully(SingleFileStore.java:67)
	at org.h2.mvstore.SingleFileStore.writeFully(SingleFileStore.java:28)
	at org.h2.mvstore.RandomAccessStore.writeChunk(RandomAccessStore.java:366)
	at org.h2.mvstore.RandomAccessStore.writeChunk(RandomAccessStore.java:28)
	at org.h2.mvstore.FileStore.storeBuffer(FileStore.java:1524)
	... 28 more
Caused by: java.nio.channels.ClosedChannelException
	at java.base/sun.nio.ch.FileChannelImpl.ensureOpen(FileChannelImpl.java:159)
	at java.base/sun.nio.ch.FileChannelImpl.write(FileChannelImpl.java:1004)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:474)
	... 33 more
2024-01-03 10:49:20.987286+05:30 database: opening C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-224]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:334)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:317)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:343)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:125)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-224]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep0(Native Method)
	at java.base/java.lang.Thread.sleep(Thread.java:484)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2024-01-03 13:25:35.284532+05:30 database: flush
org.h2.message.DbException: General error: "org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@5899fa54 failed; length 4096 at 8192 [2.2.224/2]" [50000-224]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.FileStore.storeBuffer(FileStore.java:1527)
	at org.h2.mvstore.FileStore.lambda$serializeAndStore$3(FileStore.java:1432)
	at org.h2.mvstore.FileStore.submitOrRun(FileStore.java:1401)
	at org.h2.mvstore.FileStore.serializeAndStore(FileStore.java:1432)
	at org.h2.mvstore.FileStore.lambda$storeIt$2(FileStore.java:1376)
	at org.h2.mvstore.FileStore.submitOrRun(FileStore.java:1401)
	at org.h2.mvstore.FileStore.storeIt(FileStore.java:1375)
	at org.h2.mvstore.MVStore.storeNow(MVStore.java:843)
	at org.h2.mvstore.MVStore.store(MVStore.java:808)
	at org.h2.mvstore.MVStore.commit(MVStore.java:778)
	at org.h2.mvstore.MVStore.commit(MVStore.java:770)
	at org.h2.mvstore.tx.TransactionStore.close(TransactionStore.java:365)
	at org.h2.mvstore.db.Store.close(Store.java:349)
	at org.h2.engine.Database.closeOpenFilesAndUnlock(Database.java:1293)
	at org.h2.engine.Database.<init>(Database.java:401)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:343)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:125)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@5899fa54 failed; length 4096 at 8192 [2.2.224/2]" [50000-224]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 34 more
Caused by: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@5899fa54 failed; length 4096 at 8192 [2.2.224/2]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:478)
	at org.h2.mvstore.SingleFileStore.writeFully(SingleFileStore.java:67)
	at org.h2.mvstore.SingleFileStore.writeFully(SingleFileStore.java:28)
	at org.h2.mvstore.RandomAccessStore.writeChunk(RandomAccessStore.java:366)
	at org.h2.mvstore.RandomAccessStore.writeChunk(RandomAccessStore.java:28)
	at org.h2.mvstore.FileStore.storeBuffer(FileStore.java:1524)
	... 28 more
Caused by: java.nio.channels.ClosedChannelException
	at java.base/sun.nio.ch.FileChannelImpl.ensureOpen(FileChannelImpl.java:159)
	at java.base/sun.nio.ch.FileChannelImpl.write(FileChannelImpl.java:1004)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:474)
	... 33 more
2024-01-04 09:07:37.286430+05:30 database: opening C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb
org.h2.message.DbException: IO Exception: "C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb.mv.db" [90028-224]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.mvstore.db.Store.convertMVStoreException(Store.java:168)
	at org.h2.mvstore.db.Store.createTable(Store.java:220)
	at org.h2.schema.Schema.createTable(Schema.java:797)
	at org.h2.engine.Database.<init>(Database.java:353)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:343)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:125)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: IO Exception: "C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb.mv.db" [90028-224]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 19 more
Caused by: org.h2.mvstore.MVStoreException: Reading from file sun.nio.ch.FileChannelImpl@2249f726 failed at 287304 (length -1), read 4096, remaining 0 [2.2.224/1]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.DataUtils.readFully(DataUtils.java:455)
	at org.h2.mvstore.FileStore.readFully(FileStore.java:719)
	at org.h2.mvstore.SingleFileStore.readFully(SingleFileStore.java:60)
	at org.h2.mvstore.SingleFileStore.readFully(SingleFileStore.java:28)
	at org.h2.mvstore.SFChunk.readFully(SFChunk.java:35)
	at org.h2.mvstore.Chunk.readBufferForPage(Chunk.java:465)
	at org.h2.mvstore.FileStore.readPage(FileStore.java:1960)
	at org.h2.mvstore.MVStore.readPage(MVStore.java:1021)
	at org.h2.mvstore.MVMap.readPage(MVMap.java:632)
	at org.h2.mvstore.Page$NonLeaf.getChildPage(Page.java:1117)
	at org.h2.mvstore.CursorPos.traverseDown(CursorPos.java:61)
	at org.h2.mvstore.MVMap.operate(MVMap.java:1766)
	at org.h2.mvstore.MVMap.putIfAbsent(MVMap.java:529)
	at org.h2.mvstore.tx.TransactionStore$TxMapBuilder.registerDataType(TransactionStore.java:859)
	at org.h2.mvstore.tx.TransactionStore$TxMapBuilder.create(TransactionStore.java:884)
	at org.h2.mvstore.MVStore.openMap(MVStore.java:477)
	at org.h2.mvstore.MVStore.openMap(MVStore.java:426)
	at org.h2.mvstore.tx.TransactionStore.openMap(TransactionStore.java:564)
	at org.h2.mvstore.tx.TransactionStore.openVersionedMap(TransactionStore.java:550)
	at org.h2.mvstore.tx.Transaction.openMap(Transaction.java:456)
	at org.h2.mvstore.db.MVPrimaryIndex.<init>(MVPrimaryIndex.java:53)
	at org.h2.mvstore.db.MVTable.<init>(MVTable.java:156)
	at org.h2.mvstore.db.Store.createTable(Store.java:216)
	... 16 more
Caused by: java.nio.channels.ClosedByInterruptException
	at java.base/java.nio.channels.spi.AbstractInterruptibleChannel.end(AbstractInterruptibleChannel.java:199)
	at java.base/sun.nio.ch.FileChannelImpl.endBlocking(FileChannelImpl.java:171)
	at java.base/sun.nio.ch.FileChannelImpl.readInternal(FileChannelImpl.java:994)
	at java.base/sun.nio.ch.FileChannelImpl.read(FileChannelImpl.java:966)
	at org.h2.mvstore.DataUtils.readFully(DataUtils.java:441)
	... 38 more
2024-01-04 09:07:37.513432+05:30 database: flush
org.h2.message.DbException: General error: "org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2249f726 failed; length 8192 at 57344 [2.2.224/2]" [50000-224]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.FileStore.storeBuffer(FileStore.java:1527)
	at org.h2.mvstore.FileStore.lambda$serializeAndStore$3(FileStore.java:1432)
	at org.h2.mvstore.FileStore.submitOrRun(FileStore.java:1401)
	at org.h2.mvstore.FileStore.serializeAndStore(FileStore.java:1432)
	at org.h2.mvstore.FileStore.lambda$storeIt$2(FileStore.java:1376)
	at org.h2.mvstore.FileStore.submitOrRun(FileStore.java:1401)
	at org.h2.mvstore.FileStore.storeIt(FileStore.java:1375)
	at org.h2.mvstore.MVStore.storeNow(MVStore.java:843)
	at org.h2.mvstore.MVStore.store(MVStore.java:808)
	at org.h2.mvstore.MVStore.commit(MVStore.java:778)
	at org.h2.mvstore.MVStore.commit(MVStore.java:770)
	at org.h2.mvstore.tx.TransactionStore.close(TransactionStore.java:365)
	at org.h2.mvstore.db.Store.close(Store.java:349)
	at org.h2.engine.Database.closeOpenFilesAndUnlock(Database.java:1293)
	at org.h2.engine.Database.<init>(Database.java:401)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:343)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:125)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2249f726 failed; length 8192 at 57344 [2.2.224/2]" [50000-224]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 34 more
Caused by: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2249f726 failed; length 8192 at 57344 [2.2.224/2]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:478)
	at org.h2.mvstore.SingleFileStore.writeFully(SingleFileStore.java:67)
	at org.h2.mvstore.SingleFileStore.writeFully(SingleFileStore.java:28)
	at org.h2.mvstore.RandomAccessStore.writeChunk(RandomAccessStore.java:366)
	at org.h2.mvstore.RandomAccessStore.writeChunk(RandomAccessStore.java:28)
	at org.h2.mvstore.FileStore.storeBuffer(FileStore.java:1524)
	... 28 more
Caused by: java.nio.channels.ClosedChannelException
	at java.base/sun.nio.ch.FileChannelImpl.ensureOpen(FileChannelImpl.java:159)
	at java.base/sun.nio.ch.FileChannelImpl.write(FileChannelImpl.java:1004)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:474)
	... 33 more
2024-01-04 09:07:37.888433+05:30 database: opening C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-224]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:334)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:317)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:343)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:125)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-224]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep0(Native Method)
	at java.base/java.lang.Thread.sleep(Thread.java:484)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2024-01-04 09:07:43.311104+05:30 database: CREATE CACHED TABLE "PUBLIC"."REALM_ATTRIBUTE"(
    "NAME" CHARACTER VARYING(255) NOT NULL,
    "REALM_ID" CHARACTER VARYING(36) NOT NULL,
    "VALUE" CHARACTER LARGE OBJECT
)
org.h2.jdbc.JdbcSQLNonTransientException: IO Exception: "C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb.mv.db"; SQL statement:
CREATE CACHED TABLE "PUBLIC"."REALM_ATTRIBUTE"(
    "NAME" CHARACTER VARYING(255) NOT NULL,
    "REALM_ID" CHARACTER VARYING(36) NOT NULL,
    "VALUE" CHARACTER LARGE OBJECT
) [90028-224]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.mvstore.db.Store.convertMVStoreException(Store.java:168)
	at org.h2.mvstore.db.Store.createTable(Store.java:220)
	at org.h2.schema.Schema.createTable(Schema.java:797)
	at org.h2.command.ddl.CreateTable.update(CreateTable.java:112)
	at org.h2.engine.MetaRecord.prepareAndExecute(MetaRecord.java:77)
	at org.h2.engine.Database.executeMeta(Database.java:666)
	at org.h2.engine.Database.executeMeta(Database.java:638)
	at org.h2.engine.Database.<init>(Database.java:359)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:343)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:125)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: org.h2.mvstore.MVStoreException: Reading from file sun.nio.ch.FileChannelImpl@3e3d4144 failed at 142758 (length -1), read 128, remaining 0 [2.2.224/1]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.DataUtils.readFully(DataUtils.java:455)
	at org.h2.mvstore.FileStore.readFully(FileStore.java:719)
	at org.h2.mvstore.SingleFileStore.readFully(SingleFileStore.java:60)
	at org.h2.mvstore.SingleFileStore.readFully(SingleFileStore.java:28)
	at org.h2.mvstore.SFChunk.readFully(SFChunk.java:35)
	at org.h2.mvstore.Chunk.readBufferForPage(Chunk.java:465)
	at org.h2.mvstore.FileStore.readPage(FileStore.java:1960)
	at org.h2.mvstore.MVStore.readPage(MVStore.java:1021)
	at org.h2.mvstore.MVMap.readPage(MVMap.java:632)
	at org.h2.mvstore.MVMap.readOrCreateRootPage(MVMap.java:657)
	at org.h2.mvstore.MVMap.setRootPos(MVMap.java:642)
	at org.h2.mvstore.MVStore.openMap(MVStore.java:479)
	at org.h2.mvstore.MVStore.openMap(MVStore.java:426)
	at org.h2.mvstore.tx.TransactionStore.openMap(TransactionStore.java:564)
	at org.h2.mvstore.tx.TransactionStore.openVersionedMap(TransactionStore.java:550)
	at org.h2.mvstore.tx.Transaction.openMap(Transaction.java:456)
	at org.h2.mvstore.db.MVPrimaryIndex.<init>(MVPrimaryIndex.java:53)
	at org.h2.mvstore.db.MVTable.<init>(MVTable.java:156)
	at org.h2.mvstore.db.Store.createTable(Store.java:216)
	... 20 more
Caused by: java.nio.channels.ClosedByInterruptException
	at java.base/java.nio.channels.spi.AbstractInterruptibleChannel.end(AbstractInterruptibleChannel.java:199)
	at java.base/sun.nio.ch.FileChannelImpl.endBlocking(FileChannelImpl.java:171)
	at java.base/sun.nio.ch.FileChannelImpl.readInternal(FileChannelImpl.java:994)
	at java.base/sun.nio.ch.FileChannelImpl.read(FileChannelImpl.java:966)
	at org.h2.mvstore.DataUtils.readFully(DataUtils.java:441)
	... 38 more
2024-01-04 09:07:43.425098+05:30 database: opening C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb
org.h2.message.DbException: IO Exception: "C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb.mv.db" [90028-224]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.mvstore.db.Store.convertMVStoreException(Store.java:168)
	at org.h2.mvstore.db.Store.createTable(Store.java:220)
	at org.h2.schema.Schema.createTable(Schema.java:797)
	at org.h2.command.ddl.CreateTable.update(CreateTable.java:112)
	at org.h2.engine.MetaRecord.prepareAndExecute(MetaRecord.java:77)
	at org.h2.engine.Database.executeMeta(Database.java:666)
	at org.h2.engine.Database.executeMeta(Database.java:638)
	at org.h2.engine.Database.<init>(Database.java:359)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:343)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:125)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: IO Exception: "C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb.mv.db"; SQL statement:
CREATE CACHED TABLE "PUBLIC"."REALM_ATTRIBUTE"(
    "NAME" CHARACTER VARYING(255) NOT NULL,
    "REALM_ID" CHARACTER VARYING(36) NOT NULL,
    "VALUE" CHARACTER LARGE OBJECT
) [90028-224]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 23 more
Caused by: org.h2.mvstore.MVStoreException: Reading from file sun.nio.ch.FileChannelImpl@3e3d4144 failed at 142758 (length -1), read 128, remaining 0 [2.2.224/1]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.DataUtils.readFully(DataUtils.java:455)
	at org.h2.mvstore.FileStore.readFully(FileStore.java:719)
	at org.h2.mvstore.SingleFileStore.readFully(SingleFileStore.java:60)
	at org.h2.mvstore.SingleFileStore.readFully(SingleFileStore.java:28)
	at org.h2.mvstore.SFChunk.readFully(SFChunk.java:35)
	at org.h2.mvstore.Chunk.readBufferForPage(Chunk.java:465)
	at org.h2.mvstore.FileStore.readPage(FileStore.java:1960)
	at org.h2.mvstore.MVStore.readPage(MVStore.java:1021)
	at org.h2.mvstore.MVMap.readPage(MVMap.java:632)
	at org.h2.mvstore.MVMap.readOrCreateRootPage(MVMap.java:657)
	at org.h2.mvstore.MVMap.setRootPos(MVMap.java:642)
	at org.h2.mvstore.MVStore.openMap(MVStore.java:479)
	at org.h2.mvstore.MVStore.openMap(MVStore.java:426)
	at org.h2.mvstore.tx.TransactionStore.openMap(TransactionStore.java:564)
	at org.h2.mvstore.tx.TransactionStore.openVersionedMap(TransactionStore.java:550)
	at org.h2.mvstore.tx.Transaction.openMap(Transaction.java:456)
	at org.h2.mvstore.db.MVPrimaryIndex.<init>(MVPrimaryIndex.java:53)
	at org.h2.mvstore.db.MVTable.<init>(MVTable.java:156)
	at org.h2.mvstore.db.Store.createTable(Store.java:216)
	... 20 more
Caused by: java.nio.channels.ClosedByInterruptException
	at java.base/java.nio.channels.spi.AbstractInterruptibleChannel.end(AbstractInterruptibleChannel.java:199)
	at java.base/sun.nio.ch.FileChannelImpl.endBlocking(FileChannelImpl.java:171)
	at java.base/sun.nio.ch.FileChannelImpl.readInternal(FileChannelImpl.java:994)
	at java.base/sun.nio.ch.FileChannelImpl.read(FileChannelImpl.java:966)
	at org.h2.mvstore.DataUtils.readFully(DataUtils.java:441)
	... 38 more
2024-01-04 09:07:43.584830+05:30 database: flush
org.h2.message.DbException: General error: "org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@3e3d4144 failed; length 8192 at 57344 [2.2.224/2]" [50000-224]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.FileStore.storeBuffer(FileStore.java:1527)
	at org.h2.mvstore.FileStore.lambda$serializeAndStore$3(FileStore.java:1432)
	at org.h2.mvstore.FileStore.submitOrRun(FileStore.java:1401)
	at org.h2.mvstore.FileStore.serializeAndStore(FileStore.java:1432)
	at org.h2.mvstore.FileStore.lambda$storeIt$2(FileStore.java:1376)
	at org.h2.mvstore.FileStore.submitOrRun(FileStore.java:1401)
	at org.h2.mvstore.FileStore.storeIt(FileStore.java:1375)
	at org.h2.mvstore.MVStore.storeNow(MVStore.java:843)
	at org.h2.mvstore.MVStore.store(MVStore.java:808)
	at org.h2.mvstore.MVStore.commit(MVStore.java:778)
	at org.h2.mvstore.MVStore.commit(MVStore.java:770)
	at org.h2.mvstore.tx.TransactionStore.close(TransactionStore.java:365)
	at org.h2.mvstore.db.Store.close(Store.java:349)
	at org.h2.engine.Database.closeOpenFilesAndUnlock(Database.java:1293)
	at org.h2.engine.Database.<init>(Database.java:401)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:343)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:125)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@3e3d4144 failed; length 8192 at 57344 [2.2.224/2]" [50000-224]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 34 more
Caused by: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@3e3d4144 failed; length 8192 at 57344 [2.2.224/2]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:478)
	at org.h2.mvstore.SingleFileStore.writeFully(SingleFileStore.java:67)
	at org.h2.mvstore.SingleFileStore.writeFully(SingleFileStore.java:28)
	at org.h2.mvstore.RandomAccessStore.writeChunk(RandomAccessStore.java:366)
	at org.h2.mvstore.RandomAccessStore.writeChunk(RandomAccessStore.java:28)
	at org.h2.mvstore.FileStore.storeBuffer(FileStore.java:1524)
	... 28 more
Caused by: java.nio.channels.ClosedChannelException
	at java.base/sun.nio.ch.FileChannelImpl.ensureOpen(FileChannelImpl.java:159)
	at java.base/sun.nio.ch.FileChannelImpl.write(FileChannelImpl.java:1004)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:474)
	... 33 more
2024-01-04 09:07:43.597828+05:30 database: opening C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-224]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:334)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:317)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:343)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:125)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-224]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep0(Native Method)
	at java.base/java.lang.Thread.sleep(Thread.java:484)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2024-01-04 09:52:33.906197+05:30 database: flush
org.h2.message.DbException: General error: "org.h2.mvstore.MVStoreException: Could not open file C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb.mv.db [2.2.224/1]" [50000-224]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:291)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:343)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:125)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "org.h2.mvstore.MVStoreException: Could not open file C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb.mv.db [2.2.224/1]" [50000-224]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 23 more
Caused by: org.h2.mvstore.MVStoreException: Could not open file C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb.mv.db [2.2.224/1]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:126)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	... 17 more
Caused by: java.nio.channels.ClosedByInterruptException
	at java.base/java.nio.channels.spi.AbstractInterruptibleChannel.end(AbstractInterruptibleChannel.java:199)
	at java.base/sun.nio.ch.FileChannelImpl.endBlocking(FileChannelImpl.java:171)
	at java.base/sun.nio.ch.FileChannelImpl.size(FileChannelImpl.java:430)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:120)
	... 19 more
2024-01-04 09:52:33.992113+05:30 database: opening C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb
org.h2.message.DbException: IO Exception: "C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb.mv.db" [90028-224]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.mvstore.db.Store.convertMVStoreException(Store.java:168)
	at org.h2.mvstore.db.Store.<init>(Store.java:142)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:343)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:125)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: IO Exception: "C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb.mv.db" [90028-224]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 18 more
Caused by: org.h2.mvstore.MVStoreException: Could not open file C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb.mv.db [2.2.224/1]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:126)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	... 15 more
Caused by: java.nio.channels.ClosedByInterruptException
	at java.base/java.nio.channels.spi.AbstractInterruptibleChannel.end(AbstractInterruptibleChannel.java:199)
	at java.base/sun.nio.ch.FileChannelImpl.endBlocking(FileChannelImpl.java:171)
	at java.base/sun.nio.ch.FileChannelImpl.size(FileChannelImpl.java:430)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:120)
	... 19 more
2024-01-06 13:12:10.994471+05:30 database: flush
org.h2.message.DbException: General error: "org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@6c2510c0 failed; length 8192 at 20480 [2.2.224/2]" [50000-224]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.FileStore.storeBuffer(FileStore.java:1527)
	at org.h2.mvstore.FileStore.lambda$serializeAndStore$3(FileStore.java:1432)
	at org.h2.mvstore.FileStore.submitOrRun(FileStore.java:1401)
	at org.h2.mvstore.FileStore.serializeAndStore(FileStore.java:1432)
	at org.h2.mvstore.FileStore.lambda$storeIt$2(FileStore.java:1376)
	at org.h2.mvstore.FileStore.submitOrRun(FileStore.java:1401)
	at org.h2.mvstore.FileStore.storeIt(FileStore.java:1375)
	at org.h2.mvstore.MVStore.storeNow(MVStore.java:843)
	at org.h2.mvstore.MVStore.store(MVStore.java:808)
	at org.h2.mvstore.MVStore.commit(MVStore.java:778)
	at org.h2.mvstore.MVStore.commit(MVStore.java:770)
	at org.h2.mvstore.tx.TransactionStore.close(TransactionStore.java:365)
	at org.h2.mvstore.db.Store.close(Store.java:349)
	at org.h2.engine.Database.closeOpenFilesAndUnlock(Database.java:1293)
	at org.h2.engine.Database.<init>(Database.java:401)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:343)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:125)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@6c2510c0 failed; length 8192 at 20480 [2.2.224/2]" [50000-224]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 34 more
Caused by: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@6c2510c0 failed; length 8192 at 20480 [2.2.224/2]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:478)
	at org.h2.mvstore.SingleFileStore.writeFully(SingleFileStore.java:67)
	at org.h2.mvstore.SingleFileStore.writeFully(SingleFileStore.java:28)
	at org.h2.mvstore.RandomAccessStore.writeChunk(RandomAccessStore.java:366)
	at org.h2.mvstore.RandomAccessStore.writeChunk(RandomAccessStore.java:28)
	at org.h2.mvstore.FileStore.storeBuffer(FileStore.java:1524)
	... 28 more
Caused by: java.nio.channels.ClosedChannelException
	at java.base/sun.nio.ch.FileChannelImpl.ensureOpen(FileChannelImpl.java:159)
	at java.base/sun.nio.ch.FileChannelImpl.write(FileChannelImpl.java:1004)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:474)
	... 33 more
2024-01-06 13:12:40.144314+05:30 database: flush
org.h2.message.DbException: General error: "org.h2.mvstore.MVStoreException: Could not open file C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb.mv.db [2.2.224/1]" [50000-224]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:291)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:343)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:125)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "org.h2.mvstore.MVStoreException: Could not open file C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb.mv.db [2.2.224/1]" [50000-224]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 23 more
Caused by: org.h2.mvstore.MVStoreException: Could not open file C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb.mv.db [2.2.224/1]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:126)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	... 17 more
Caused by: java.nio.channels.ClosedByInterruptException
	at java.base/java.nio.channels.spi.AbstractInterruptibleChannel.end(AbstractInterruptibleChannel.java:199)
	at java.base/sun.nio.ch.FileChannelImpl.endBlocking(FileChannelImpl.java:171)
	at java.base/sun.nio.ch.FileChannelImpl.size(FileChannelImpl.java:430)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:120)
	... 19 more
2024-01-06 13:12:40.342698+05:30 database: opening C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb
org.h2.message.DbException: IO Exception: "C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb.mv.db" [90028-224]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.mvstore.db.Store.convertMVStoreException(Store.java:168)
	at org.h2.mvstore.db.Store.<init>(Store.java:142)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:343)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:125)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: IO Exception: "C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb.mv.db" [90028-224]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 18 more
Caused by: org.h2.mvstore.MVStoreException: Could not open file C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb.mv.db [2.2.224/1]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:126)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	... 15 more
Caused by: java.nio.channels.ClosedByInterruptException
	at java.base/java.nio.channels.spi.AbstractInterruptibleChannel.end(AbstractInterruptibleChannel.java:199)
	at java.base/sun.nio.ch.FileChannelImpl.endBlocking(FileChannelImpl.java:171)
	at java.base/sun.nio.ch.FileChannelImpl.size(FileChannelImpl.java:430)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:120)
	... 19 more
2024-01-06 13:57:10.869608+05:30 database: CREATE CACHED TABLE "PUBLIC"."CLIENT_SCOPE"(
    "ID" CHARACTER VARYING(36) NOT NULL,
    "NAME" CHARACTER VARYING(255),
    "REALM_ID" CHARACTER VARYING(36),
    "DESCRIPTION" CHARACTER VARYING(255),
    "PROTOCOL" CHARACTER VARYING(255)
)
org.h2.jdbc.JdbcSQLNonTransientException: IO Exception: "C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb.mv.db"; SQL statement:
CREATE CACHED TABLE "PUBLIC"."CLIENT_SCOPE"(
    "ID" CHARACTER VARYING(36) NOT NULL,
    "NAME" CHARACTER VARYING(255),
    "REALM_ID" CHARACTER VARYING(36),
    "DESCRIPTION" CHARACTER VARYING(255),
    "PROTOCOL" CHARACTER VARYING(255)
) [90028-224]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.mvstore.db.Store.convertMVStoreException(Store.java:168)
	at org.h2.mvstore.db.Store.createTable(Store.java:220)
	at org.h2.schema.Schema.createTable(Schema.java:797)
	at org.h2.command.ddl.CreateTable.update(CreateTable.java:112)
	at org.h2.engine.MetaRecord.prepareAndExecute(MetaRecord.java:77)
	at org.h2.engine.Database.executeMeta(Database.java:666)
	at org.h2.engine.Database.executeMeta(Database.java:638)
	at org.h2.engine.Database.<init>(Database.java:359)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:343)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:125)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: org.h2.mvstore.MVStoreException: Reading from file sun.nio.ch.FileChannelImpl@7967cd5b failed at 1356905 (length -1), read 32, remaining 0 [2.2.224/1]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.DataUtils.readFully(DataUtils.java:455)
	at org.h2.mvstore.FileStore.readFully(FileStore.java:719)
	at org.h2.mvstore.SingleFileStore.readFully(SingleFileStore.java:60)
	at org.h2.mvstore.SingleFileStore.readFully(SingleFileStore.java:28)
	at org.h2.mvstore.SFChunk.readFully(SFChunk.java:35)
	at org.h2.mvstore.Chunk.readBufferForPage(Chunk.java:465)
	at org.h2.mvstore.FileStore.readPage(FileStore.java:1960)
	at org.h2.mvstore.MVStore.readPage(MVStore.java:1021)
	at org.h2.mvstore.MVMap.readPage(MVMap.java:632)
	at org.h2.mvstore.MVMap.readOrCreateRootPage(MVMap.java:657)
	at org.h2.mvstore.MVMap.setRootPos(MVMap.java:642)
	at org.h2.mvstore.MVStore.openMap(MVStore.java:479)
	at org.h2.mvstore.MVStore.openMap(MVStore.java:426)
	at org.h2.mvstore.tx.TransactionStore.openMap(TransactionStore.java:564)
	at org.h2.mvstore.tx.TransactionStore.openVersionedMap(TransactionStore.java:550)
	at org.h2.mvstore.tx.Transaction.openMap(Transaction.java:456)
	at org.h2.mvstore.db.MVPrimaryIndex.<init>(MVPrimaryIndex.java:53)
	at org.h2.mvstore.db.MVTable.<init>(MVTable.java:156)
	at org.h2.mvstore.db.Store.createTable(Store.java:216)
	... 20 more
Caused by: java.nio.channels.ClosedByInterruptException
	at java.base/java.nio.channels.spi.AbstractInterruptibleChannel.end(AbstractInterruptibleChannel.java:199)
	at java.base/sun.nio.ch.FileChannelImpl.endBlocking(FileChannelImpl.java:171)
	at java.base/sun.nio.ch.FileChannelImpl.readInternal(FileChannelImpl.java:994)
	at java.base/sun.nio.ch.FileChannelImpl.read(FileChannelImpl.java:966)
	at org.h2.mvstore.DataUtils.readFully(DataUtils.java:441)
	... 38 more
2024-01-06 13:57:11.147609+05:30 database: opening C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb
org.h2.message.DbException: IO Exception: "C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb.mv.db" [90028-224]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.mvstore.db.Store.convertMVStoreException(Store.java:168)
	at org.h2.mvstore.db.Store.createTable(Store.java:220)
	at org.h2.schema.Schema.createTable(Schema.java:797)
	at org.h2.command.ddl.CreateTable.update(CreateTable.java:112)
	at org.h2.engine.MetaRecord.prepareAndExecute(MetaRecord.java:77)
	at org.h2.engine.Database.executeMeta(Database.java:666)
	at org.h2.engine.Database.executeMeta(Database.java:638)
	at org.h2.engine.Database.<init>(Database.java:359)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:343)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:125)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: IO Exception: "C:/Users/savadr/Desktop/Projects-Learnings-/keycloak-23.0.3/data/h2/keycloakdb.mv.db"; SQL statement:
CREATE CACHED TABLE "PUBLIC"."CLIENT_SCOPE"(
    "ID" CHARACTER VARYING(36) NOT NULL,
    "NAME" CHARACTER VARYING(255),
    "REALM_ID" CHARACTER VARYING(36),
    "DESCRIPTION" CHARACTER VARYING(255),
    "PROTOCOL" CHARACTER VARYING(255)
) [90028-224]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 23 more
Caused by: org.h2.mvstore.MVStoreException: Reading from file sun.nio.ch.FileChannelImpl@7967cd5b failed at 1356905 (length -1), read 32, remaining 0 [2.2.224/1]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.DataUtils.readFully(DataUtils.java:455)
	at org.h2.mvstore.FileStore.readFully(FileStore.java:719)
	at org.h2.mvstore.SingleFileStore.readFully(SingleFileStore.java:60)
	at org.h2.mvstore.SingleFileStore.readFully(SingleFileStore.java:28)
	at org.h2.mvstore.SFChunk.readFully(SFChunk.java:35)
	at org.h2.mvstore.Chunk.readBufferForPage(Chunk.java:465)
	at org.h2.mvstore.FileStore.readPage(FileStore.java:1960)
	at org.h2.mvstore.MVStore.readPage(MVStore.java:1021)
	at org.h2.mvstore.MVMap.readPage(MVMap.java:632)
	at org.h2.mvstore.MVMap.readOrCreateRootPage(MVMap.java:657)
	at org.h2.mvstore.MVMap.setRootPos(MVMap.java:642)
	at org.h2.mvstore.MVStore.openMap(MVStore.java:479)
	at org.h2.mvstore.MVStore.openMap(MVStore.java:426)
	at org.h2.mvstore.tx.TransactionStore.openMap(TransactionStore.java:564)
	at org.h2.mvstore.tx.TransactionStore.openVersionedMap(TransactionStore.java:550)
	at org.h2.mvstore.tx.Transaction.openMap(Transaction.java:456)
	at org.h2.mvstore.db.MVPrimaryIndex.<init>(MVPrimaryIndex.java:53)
	at org.h2.mvstore.db.MVTable.<init>(MVTable.java:156)
	at org.h2.mvstore.db.Store.createTable(Store.java:216)
	... 20 more
Caused by: java.nio.channels.ClosedByInterruptException
	at java.base/java.nio.channels.spi.AbstractInterruptibleChannel.end(AbstractInterruptibleChannel.java:199)
	at java.base/sun.nio.ch.FileChannelImpl.endBlocking(FileChannelImpl.java:171)
	at java.base/sun.nio.ch.FileChannelImpl.readInternal(FileChannelImpl.java:994)
	at java.base/sun.nio.ch.FileChannelImpl.read(FileChannelImpl.java:966)
	at org.h2.mvstore.DataUtils.readFully(DataUtils.java:441)
	... 38 more
2024-01-06 13:57:11.191194+05:30 database: flush
org.h2.message.DbException: General error: "org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@7967cd5b failed; length 8192 at 28672 [2.2.224/2]" [50000-224]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.FileStore.storeBuffer(FileStore.java:1527)
	at org.h2.mvstore.FileStore.lambda$serializeAndStore$3(FileStore.java:1432)
	at org.h2.mvstore.FileStore.submitOrRun(FileStore.java:1401)
	at org.h2.mvstore.FileStore.serializeAndStore(FileStore.java:1432)
	at org.h2.mvstore.FileStore.lambda$storeIt$2(FileStore.java:1376)
	at org.h2.mvstore.FileStore.submitOrRun(FileStore.java:1401)
	at org.h2.mvstore.FileStore.storeIt(FileStore.java:1375)
	at org.h2.mvstore.MVStore.storeNow(MVStore.java:843)
	at org.h2.mvstore.MVStore.store(MVStore.java:808)
	at org.h2.mvstore.MVStore.commit(MVStore.java:778)
	at org.h2.mvstore.MVStore.commit(MVStore.java:770)
	at org.h2.mvstore.tx.TransactionStore.close(TransactionStore.java:365)
	at org.h2.mvstore.db.Store.close(Store.java:349)
	at org.h2.engine.Database.closeOpenFilesAndUnlock(Database.java:1293)
	at org.h2.engine.Database.<init>(Database.java:401)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:343)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:125)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@7967cd5b failed; length 8192 at 28672 [2.2.224/2]" [50000-224]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 34 more
Caused by: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@7967cd5b failed; length 8192 at 28672 [2.2.224/2]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:478)
	at org.h2.mvstore.SingleFileStore.writeFully(SingleFileStore.java:67)
	at org.h2.mvstore.SingleFileStore.writeFully(SingleFileStore.java:28)
	at org.h2.mvstore.RandomAccessStore.writeChunk(RandomAccessStore.java:366)
	at org.h2.mvstore.RandomAccessStore.writeChunk(RandomAccessStore.java:28)
	at org.h2.mvstore.FileStore.storeBuffer(FileStore.java:1524)
	... 28 more
Caused by: java.nio.channels.ClosedChannelException
	at java.base/sun.nio.ch.FileChannelImpl.ensureOpen(FileChannelImpl.java:159)
	at java.base/sun.nio.ch.FileChannelImpl.write(FileChannelImpl.java:1004)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:474)
	... 33 more
